components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -237,107 +246,121 @@ public void registerOAuthApplicationData(OAuthConsumerAppDTO application) throws
     public OAuthConsumerAppDTO registerAndRetrieveOAuthApplicationData(OAuthConsumerAppDTO application)
             throws IdentityOAuthAdminException {
 
-        String tenantAwareLoggedInUser = CarbonContext.getThreadLocalCarbonContext().getUsername();
+        String tenantAwareLoggedInUsername = CarbonContext.getThreadLocalCarbonContext().getUsername();
+        String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
         OAuthAppDO app = new OAuthAppDO();
-        if (tenantAwareLoggedInUser != null) {
-            String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        AuthenticatedUser defaultAppOwner = null;
+        try {
+            if (tenantAwareLoggedInUsername != null) {
```

Comment:
Shall we use `StringUtils.isNotEmpty(String str)`, since it will inspect for both empty and null check

Reply:
resolved by 57676650f4bb2e0fb744eae58ff5aebe4aa65b2f